### PR TITLE
Add socketsio listeners

### PR DESCRIPTION
**Pull request by bobobob22**

### CODE REVIEW

File: packages/roomSync/@types/socket.io-client/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+declare module 'socket.io-client' {
```
Comment (kowalski) [inline packages/roomSync/@types/socket.io-client/index.d.ts:1]: didn't we already move types to common location ?

File: packages/roomSync/package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "redux-saga": "^1.1.3",
     "redux-saga-test-plan": "4.0.1",
     "reselect": "4.0.0",
+    "socket.io-client": "^3.0.4",
```
Comment (kowalski) [inline packages/roomSync/package.json:38]: this should be a dependcy of signal-server not roomSync

File: packages/roomSync/src/store/djSeats/djSeats.saga.ts [hunk 3]
```diff
@@ -7,13 +7,16 @@ import { djSeatsSelectors } from './djSeats.selector'
 import { DjSeat } from './djSeats.types'
 
 export function* onTakeDjSeatMessage(action: DjSeatsActions['onTakeDjSeatMessage']): Generator {
+  const djSeats = yield select(djSeatsSelectors.djSeats)
```
Comment (kowalski) [inline packages/roomSync/src/store/djSeats/djSeats.saga.ts:10]: you're not using it apart from console.log below

File: packages/roomSync/src/store/djSeats/djSeats.saga.ts [hunk 4]
```diff
@@ -7,13 +7,16 @@ import { djSeatsSelectors } from './djSeats.selector'
...
   const djSeat = ((yield select(djSeatsSelectors.djSeats)) as Map<number, DjSeat>).get(action.payload.djSeatKey)
   if (!djSeat) {
     console.error('No provided djSeat')
 
     return
   }
 
+  console.log(djSeats, djSeat)
```
Comment (kowalski) [inline packages/roomSync/src/store/djSeats/djSeats.saga.ts:18]: console.log

File: packages/roomSync/src/store/djSeats/djSeats.saga.ts [hunk 5]
```diff
@@ -24,6 +27,7 @@ export function* onTakeDjSeatMessage(action: DjSeatsActions['onTakeDjSeatMessage
   } as DjSeat
   const isCurrentSeatValidCheck = yield call(isCurrentSeatValid, djSeat, newDjSeat)
 
+  console.log('')
```
Comment (kowalski) [inline packages/roomSync/src/store/djSeats/djSeats.saga.ts:30]: console.log

File: packages/signaling-server/src/SocketIoConnection/SocketIoConnection.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+  public static getInstance(socket, io): SocketIoConnection {
+    return this.instance || (this.instance = new this(socket, io))
+  }
+
+  public emitUserDisconnect(): void {
+    this.io.emit("userDisconnect", )
+  }
+
+  public emitLeaveDjSeat(data) {
```
Comment (kowalski) [inline packages/signaling-server/src/SocketIoConnection/SocketIoConnection.ts:35]: typing for `data` would be welcome (for this and all following methods in this class)

File: packages/signaling-server/src/helpers/configureListenersForRoomBroadcastSocketIo.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import { UsersType } from '../types/types'
+
+import {
+  IAddAvatarToDancefloorBroadcast, IInitState,
+  ILeaveDjSeat,
+  ISendMessage,
+  ITakeDjSeat,
+} from '@turntable/roomSync/store/broadcast/broadcast.reducer'
+
+export const configureListenersForRoomBroadcast = (
```
Comment (kowalski) [inline packages/signaling-server/src/helpers/configureListenersForRoomBroadcastSocketIo.ts:15]: You are calling this method for every established connection.

That's not the way I suggested.

I suggested that you have a class that keeps track of all the connections and dispatches the broadcasts to all connections.
This way you don't need to care about cleanup of these listeners when connection goes away (which you didn't do btw)

File: packages/signaling-server/src/index.ts [hunk 8]
```diff
@@ -1,184 +1,64 @@
...
+    socket.on('new-user', (name) => {
+      users[socket.id] = name
+      socket.broadcast.emit('user-connected', name)
+    })
 
-  app.get('/', (req, res) => {
-    console.log(JSON.stringify(store.getState(), replacer, 2))
-    res.send(JSON.stringify(store.getState(), replacer, 2))
+    const connectedSocket = SocketIoConnection.getInstance(socket, io)
+    await configureListenersForRoomBroadcast(roomBroadcast, store, connectedSocket, users)
```
Comment (kowalski) [inline packages/signaling-server/src/index.ts:56]: we should add listeners for every connection. Just add one listener and make it responsible for dispatching events to all socket.io connections

File: packages/signaling-server/src/index.ts [hunk 9]
```diff
@@ -1,184 +1,64 @@
...
   })
 
-  app.listen(port, () => {
-    console.log(`Example app listening at http://localhost:${port}`)
-  })
+  server.listen(port, () => console.log(`Listening on port ${port}`))
 
+  app.get('/', (request, response) => {
```
Comment (kowalski) [inline packages/signaling-server/src/index.ts:61]: Please be sure to remove this after we are done with debugging

File: packages/signaling-server/src/store/configureStore.ts [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+import { keyNames } from '@turntable/roomSync/store/consts/storeKeys'
+import { playerReducer } from '@turntable/roomSync/store/player/player.reducer'
+import { djSeatsReducer } from '@turntable/roomSync/store/djSeats/djSeats.reducer'
+import { initStateReducer } from '@turntable/roomSync/store/initState/initState.reducer'
+import { dancefloorReducer } from '@turntable/roomSync/store/dancefloor/dancefloor.reducer'
+import { peerIdReducer } from '@turntable/roomSync/store/peerId/peerId.reducer'
+
+import { Store, applyMiddleware, combineReducers, createStore, Reducer } from '@reduxjs/toolkit'
+import createSagaMiddleware from 'redux-saga'
+import { all as effectsAll, fork as effectsFork } from 'redux-saga/effects'
```
Comment (kowalski) [inline packages/signaling-server/src/store/configureStore.ts:17]: You importing here as `effectAll` and `effectsFork` to just below rebind it using

```

const all: any = effectsAll
const fork: any = effectsFork```

Maybe use import it as `all` and `fork` ?

File: packages/socketIoClient/src/index.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+"use strict";
```
Comment (kowalski) [inline packages/socketIoClient/src/index.js:1]: probably don't add `.js` file to git. `git ignore` it

File: packages/types/package.json [hunk 12]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@turntable/types",
```
Comment (kowalski) [inline packages/types/package.json:2]: ah, ok, so you're adding the types package here, good

So please remove all the instances of it which are duplicated in each of the packages (webapp, signaling-server, etc)

**Review by kowalski — CHANGES_REQUESTED** [review]

File: packages/roomSync/package.json (hunk 2 above)
Comment (bobobob22) [reply packages/roomSync/package.json:38]: ok

File: packages/roomSync/src/store/djSeats/djSeats.saga.ts (hunk 4 above)
Comment (bobobob22) [reply packages/roomSync/src/store/djSeats/djSeats.saga.ts]: ok

File: packages/roomSync/src/store/djSeats/djSeats.saga.ts (hunk 5 above)
Comment (bobobob22) [reply packages/roomSync/src/store/djSeats/djSeats.saga.ts]: ok

File: packages/signaling-server/src/store/configureStore.ts (hunk 10 above)
Comment (bobobob22) [reply packages/signaling-server/src/store/configureStore.ts:17]: ok

File: packages/signaling-server/src/SocketIoConnection/SocketIoConnection.ts (hunk 6 above)
Comment (bobobob22) [reply packages/signaling-server/src/SocketIoConnection/SocketIoConnection.ts]: ok

File: packages/socketIoClient/src/index.js (hunk 11 above)
Comment (bobobob22) [reply packages/socketIoClient/src/index.js]: ok

File: packages/roomSync/@types/socket.io-client/index.d.ts (hunk 1 above)
Comment (bobobob22) [reply packages/roomSync/@types/socket.io-client/index.d.ts]: ok

File: packages/signaling-server/src/index.ts (hunk 8 above)
Comment (bobobob22) [reply packages/signaling-server/src/index.ts]: ok

File: packages/types/package.json (hunk 12 above)
Comment (bobobob22) [reply packages/types/package.json:2]: ok

File: packages/roomSync/src/store/djSeats/djSeats.saga.ts (hunk 3 above)
Comment (bobobob22) [reply packages/roomSync/src/store/djSeats/djSeats.saga.ts]: It is used below when djSeat exists in `const newDjSeat{}`

File: packages/signaling-server/src/index.ts (hunk 9 above)
Comment (bobobob22) [reply packages/signaling-server/src/index.ts:48]: ok

File: packages/signaling-server/src/helpers/configureListenersForRoomBroadcastSocketIo.ts (hunk 7 above)
Comment (bobobob22) [reply packages/signaling-server/src/helpers/configureListenersForRoomBroadcastSocketIo.ts:13]: ok

**Pull request merged**